### PR TITLE
ACS-2259 Documentation for date math in range queries.

### PR DESCRIPTION
**Pull request by tpage-alfresco**

### CODE REVIEW

**Review by mjstanford — APPROVED** [review]

**Comment by tpage-alfresco** [comment]
This relates to https://alfresco.atlassian.net/browse/DOCS-6513

File: search-enterprise/latest/using/index.md [hunk 1]
```diff
@@ -230,6 +230,28 @@ In the REST API you can specify the timezone to be used in search for date range
...
 ```
 
+### Search using date math
+
+Date range queries can be even more powerful when applying date math functions. AFTS supports adding and subtracting periods, as well as rounding:
+
+| AFTS query | Description | Notes |
+| ---------- | ----------- | ----- |
+| `acme:projectStartDate:[NOW TO NOW+1DAY>` | Documents with a project start date in the next twenty four hours | |
+| `acme:projectStartDate:[NOW/DAY TO NOW/DAY+1DAY>` | Documents with a project start date today (between midnight and midnight UTC) | Note the subtle difference with the previous query |
```
Comment (oliveralfresco2018) [inline search-enterprise/latest/using/index.md:240]: Hello @tpage-alfresco thanks for this. When you say between midnight and midnight UTC, what is the timezone of the first midnight?

Cheers

File: search-enterprise/latest/using/index.md [hunk 2]
```diff
@@ -230,6 +230,28 @@ In the REST API you can specify the timezone to be used in search for date range
...
+| `acme:projectStartDate:[NOW-1MONTH/YEAR TO NOW-1MONTH/YEAR+1DAY>` | Documents with a project start date in the first day of the year that last month fell in | It's possible to chain date math functions together |
```
[the remark this answers is not in the document]
Comment (tpage-alfresco) [reply search-enterprise/latest/using/index.md:241]: As in the sentence "Easter fell in April last year"

File: search-enterprise/latest/using/index.md (hunk 1 above)
Comment (tpage-alfresco) [reply search-enterprise/latest/using/index.md:240]: Also UTC: "between (midnight and midnight) UTC"

File: search-enterprise/latest/using/index.md [hunk 3]
```diff
@@ -232,23 +232,24 @@ In the REST API you can specify the timezone to be used in search for date range
...
-| AFTS query | Description | Notes |
-| ---------- | ----------- | ----- |
-| `acme:projectStartDate:[NOW TO NOW+1DAY>` | Documents with a project start date in the next twenty four hours | |
-| `acme:projectStartDate:[NOW/DAY TO NOW/DAY+1DAY>` | Documents with a project start date today (between midnight and midnight UTC) | Note the subtle difference with the previous query |
-| `acme:projectStartDate:[NOW-1MONTH/YEAR TO NOW-1MONTH/YEAR+1DAY>` | Documents with a project start date in the first day of the year that last month fell in | It's possible to chain date math functions together |
-| `cm:created:[2020-11-01T12:34:00/YEAR TO NOW>` | Documents created since the start of 2020. | It's also possible to apply date math to absolute points in time. |
+| AFTS query | Description |
+| ---------- | ----------- |
+| `acme:projectStartDate:[NOW TO NOW+1DAY>` | Documents that have a project start date in the next twenty four hours. |
+| `acme:projectStartDate:[NOW/DAY TO NOW/DAY+1DAY>` | Documents that have a project start date today and between midnight and midnight (UTC), **Note:** The minor difference between this query and the one above. |
```
Comment (tpage-alfresco) [inline search-enterprise/latest/using/index.md:240]: > Documents that have a project start date today and between midnight and midnight (UTC)
The two clauses are identical - "between midnight and midnight UTC" is just a clarification of "today"

File: search-enterprise/latest/using/index.md [hunk 4]
```diff
@@ -232,23 +232,24 @@ In the REST API you can specify the timezone to be used in search for date range
...
+| `acme:projectStartDate:[NOW-1MONTH/YEAR TO NOW-1MONTH/YEAR+1DAY>` | Documents with a project start date in the first day of the year that last month fell in
+documents with a project start date in (a day). **Note:** It's possible to chain date math functions at the same time. |
```
Comment (tpage-alfresco) [inline search-enterprise/latest/using/index.md:242]: I don't understand this change - you seem to have added half a sentence? ("documents with a project start date in (a day)")

Comment (tpage-alfresco) [reply search-enterprise/latest/using/index.md:242]: I'm not sure the note makes sense either.

File: search-enterprise/latest/using/index.md [hunk 5]
```diff
@@ -232,23 +232,24 @@ In the REST API you can specify the timezone to be used in search for date range
...
+| `cm:created:[2020-11-01T12:34:00/YEAR TO NOW>` | Documents that were created from the the start of 2020. **Note:** It's also possible to apply date math to absolute points in time. |
```
Comment (tpage-alfresco) [inline search-enterprise/latest/using/index.md:243]: "from" -> "since"

**Review by tpage-alfresco — COMMENTED** [review]
I think we still need to revisit a few of these sentences.

File: search-enterprise/latest/using/index.md (hunk 3 above)
Comment (oliveralfresco2018) [reply search-enterprise/latest/using/index.md:240]: | `acme:projectStartDate:[NOW/DAY TO NOW/DAY+1DAY>` | Documents that have a project start date from the current day. The current day is defined as from midnight to midnight (UTC), **Note:** The minor difference between this query and the one above. |

File: search-enterprise/latest/using/index.md (hunk 4 above)
Comment (oliveralfresco2018) [reply search-enterprise/latest/using/index.md:242]: Hi Tom, 

Thanks. I made the change based on what we spoke about on Slack. 
Can you provide with a clear sentence of what you need? 
The note is from the original change and provided to me, if you're not happy with it let me know something that is more suitable.
Cheers

File: search-enterprise/latest/using/index.md (hunk 3 above)
Comment (tpage-alfresco) [reply search-enterprise/latest/using/index.md:240]: I don't think the difference between the two rows is "minor" - I think it's "subtle".  You could just remove the word "minor" completely if you don't want to use "subtle" though.

Comment (tpage-alfresco) [reply search-enterprise/latest/using/index.md:240]: "Documents that have a project start date from the current day" -> "Documents that have a project start date in the current day"

File: search-enterprise/latest/using/index.md (hunk 4 above)
Comment (tpage-alfresco) [reply search-enterprise/latest/using/index.md:242]: I prefer the note that I wrote: "It's possible to chain date math functions together"

The first sentence no longer makes sense at the moment - I'm not sure it's even a single sentence:
> Documents with a project start date in the first day of the year that last month fell in
> documents with a project start date in (a day).

I originally wrote:
> Documents with a project start date in the first day of the year that last month fell in

but you didn't find this easy to read.  Perhaps this is clearer?
> Documents with a project start date in the first day of this year, or in the first day of last year if it is currently January.

File: search-enterprise/latest/using/index.md [hunk 6]
```diff
@@ -232,23 +232,23 @@ In the REST API you can specify the timezone to be used in search for date range
...
-| AFTS query | Description | Notes |
-| ---------- | ----------- | ----- |
-| `acme:projectStartDate:[NOW TO NOW+1DAY>` | Documents with a project start date in the next twenty four hours | |
-| `acme:projectStartDate:[NOW/DAY TO NOW/DAY+1DAY>` | Documents with a project start date today (between midnight and midnight UTC) | Note the subtle difference with the previous query |
-| `acme:projectStartDate:[NOW-1MONTH/YEAR TO NOW-1MONTH/YEAR+1DAY>` | Documents with a project start date in the first day of the year that last month fell in | It's possible to chain date math functions together |
-| `cm:created:[2020-11-01T12:34:00/YEAR TO NOW>` | Documents created since the start of 2020. | It's also possible to apply date math to absolute points in time. |
+| AFTS query | Description |
+| ---------- | ----------- |
+| `acme:projectStartDate:[NOW TO NOW+1DAY>` | Documents that have a project start date in the next twenty four hours. |
+| `acme:projectStartDate:[NOW/DAY TO NOW/DAY+1DAY>` | Documents that have a project start date from the current day. The current day is defined as from midnight to midnight (UTC), **Note:** The minor difference between this query and the one above. |
```
Comment (tpage-alfresco) [inline search-enterprise/latest/using/index.md:240]: I don't think this is a "minor" difference - IMO it's "subtle".

**Comment by tpage-alfresco** [comment]
LGTM. I'm not able to approve since I created the PR, but here's a virtual :+1: from me.

**Review by mjstanford — APPROVED** [review]

**Review by gravitonian — APPROVED** [review]
LGTM

**Pull request merged**